connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -229,6 +229,10 @@ private Long ensureLeaderConfig(long maxOffset, WorkerCoordinator coordinator) {
         Map<String, ConnectorsAndTasks> toRevoke = computeDeleted(deleted, connectorAssignments, taskAssignments);
         log.debug("Connector and task to delete assignments: {}", toRevoke);
 
+        // It may cause some duplicated assignment, when a worker with old generation assignment re-join the group
```

Comment:
The comment is a bit unclear. This code is not causing the duplicated assignment. Is resolving it by revoking redundant tasks. Can we rephrase?